core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -371,6 +372,14 @@ else if (taskHolder.getTaskExecution() != null) {
             fullGcTime = taskContext.getFullGcTime();
             dynamicFiltersVersion = taskContext.getDynamicFiltersVersion();
         }
+        else {
+            if (state == FINISHED) {
+                // if task FINISHED successfully but taskHolder is not yet updated with SqlTaskExecution or FinalTaskInfo
+                // we are masking the state and return RUNNING. This is important so coordinator would not consider incomplete
+                // task information (e.g. missing proper dynamicFiltersVersion as final)
+                state = RUNNING;
```

Comment:
maybe include some advertisement that this won't introduce much delay, since the transition to having SqlTaskExecution should happen immediately